Makefile

```diff
@@ -52,6 +52,21 @@ endif
 export GOBIN := $(BIN_DIR)
 GOTESTSUM := $(GOBIN)/gotestsum
 
+BUILD_DEPS := $(BIN_DIR)/libgpu-setup.so
+export CGO_LDFLAGS := -L$(BIN_DIR)  -Wl,-rpath,$(BIN_DIR)
+
+# We depend on post submodule Makefile.Inc fetching
+# libgpu-setup.so so there is no must
+# to keep it in sync with Go's dependency on post package.
+post/Makefile.Inc:
```

Comment:
Why not copy the Makefile instead of having the whole repository as submodule?

Reply:
It would work too. I don't think that the performance of cloning will be an issue here. It is cloned only once after all.

For me, a git submodule is a natural tool for this kind of problem. I'm curious about what others think. @countvonzero @dshulyak  @moshababo?

Reply:
@fasmat Let's close on this ASAP. IMO it's not worth delaying the merge because of this. Are you OK to merge it with the git submodule and discuss it offline? Or I can change to the approach go-spacemesh took (to just put Makefile.Inc in the repo). I'm fine with either option.

Reply:
lets use the copy approach; how to compile poet should be in poet not in gpu-post

Reply:
Will do. However, I think that how to build **post** should be handled by post so that the code that depends on it doesn't need to know how to build the C library that post depends on.